### ISSUE TITLE
Autocommand not working with ChkTeX

### ISSUE THREAD

**Issue by tarsobcaldas** [mfussenegger/nvim-lint#162]
I'm trying to get the autocommand to work in my lua config with
```lua
require('lint').linters_by_ft = {
	tex = {'chktex'}
}
```

and

```lua
vim.cmd([[au BufWritePost <buffer> lua require('lint').try_lint()]]) 
```

but it's not working. When I run only `lua require('lint').try_lint` on a .tex buffer, it works, but the autocmd doesn't. 

I'm using Windows 11 with Neovim 6.0. I'm trying to build a lua only neovim config, so I'm not using .vimrc or any .vim file. Will try though, if anything else fails.

**Comment by mfussenegger** [comment]
Where do you run `vim.cmd([[au BufWritePost <buffer> lua require('lint').try_lint()]])` ? If it is only set once in the `init.lua` it will only be triggered once on initial startup. The `<buffer>` in the autocmd definition means it will only apply for the current  buffer at the time the `vim.cmd` call is made.

You can either remove the `<buffer>` or make sure you setup the `BufWritePost` autocmd on each buffer

**Comment by fillipe-gsm** [comment]
I was facing the same issue but with `pylint`. When writing

```lua
vim.cmd([[ au BufWritePost <buffer> lua require('lint').try_lint() ]])
```

in my `init.lua` the linter was never triggered, but when I ran `au BufWritePost <buffer> lua require('lint').try_lint()` manually at an opened file it worked.

Following @mfussenegger's explanation, I replaced the previous line with

```lua
vim.cmd([[ au BufWritePost * lua require('lint').try_lint() ]])
```

and now the linter is automatically called whenever I write at any buffer.
